src/main/java/ch/njol/skript/expressions/ExprRandomCharacter.java

```diff
@@ -0,0 +1,130 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Random;
+
+@Name("Random Character")
+@Description({
+	"One or more random characters between two given characters. Use 'alphanumeric' if you want only alphanumeric characters.",
+	"This expression uses the Unicode numerical code of a character to determine which characters are between the two given characters.",
+	"If strings of more than one character are given, only the first character of each is used."
+})
+@Examples({
+	"set {_captcha} to join (5 random characters between \"a\" and \"z\") with \"\"",
+	"send 3 random alphanumeric characters between \"0\" and \"z\""
+})
+@Since("INSERT VERSION")
+public class ExprRandomCharacter extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprRandomCharacter.class, String.class, ExpressionType.COMBINED,
+				"[a|%-number%] random [:alphanumeric] character[s] (from|between) %string% (to|and) %string%");
+	}
+
+	@Nullable
+	private Expression<Number> amount;
+	private Expression<String> from, to;
+	private boolean isAlphanumeric;
+	private final Random rand = new Random();
```

Comment:
Do you think it's worth creating a new Random seed on every call for more randomness or will it be so heavy and not worth it?

Reply:
I took my cues from ExprRandomNumber, which does it like this. Personally I don't think there'd be much benefit, but I also don't know too much about Random.

Reply:
Gotcha, well it might not be a big deal for many but I am positive it will make some difference in the results but at the cost of performance (not much I think), I threw a question in [skriptlang-developement](https://discord.com/channels/135877399391764480/836220422223036467/1135482211567665202) for more thoughts on this